test/unit/knex.js

```diff
@@ -310,6 +310,40 @@ describe('knex', () => {
     });
   });
 
+  it('propagates error correctly when all connections are in use', function() {
+    this.timeout(30000);
```

Comment:
Why this needs 30secs timeout? 🤔

Reply:
It is unlikely to use any more than just a couple of seconds, it is going to crash and be handled way earlier than that.